paddlenlp/transformers/ctrl/modeling.py

```diff
@@ -0,0 +1,764 @@
+        return x.transpose(perm=[0, 2, 1, 3])
+
+    def forward(
+            self,
+            v,
+            k,
+            q,
+            mask,
+            layer_past=None,
+            attention_mask=None,
+            use_cache=False,
+            output_attentions=False, ):
+        batch_size = q.shape[0]
+
+        q = self.Wq(q)
+        k = self.Wk(k)
+        v = self.Wv(v)
+
+        q = self.split_into_heads(q, batch_size)
+        k = self.split_into_heads(k, batch_size)
+        v = self.split_into_heads(v, batch_size)
+        if layer_past is not None:
+            past_key, past_value = layer_past[0], layer_past[1]
+            k = paddle.concat([past_key, k], axis=-2)
+            v = paddle.concat([past_value, v], axis=-2)
+
+        if use_cache is True:
+            present = paddle.stack([k, v])
+        else:
+            present = (None, )
+
+        scaled_attention, attn = scaled_dot_product_attention(q, k, v, mask,
+                                                              attention_mask)
+        scaled_attention = scaled_attention.transpose([0, 2, 1, 3])
+
+        original_size_attention = scaled_attention.reshape(
+            shape=[batch_size, -1, self.d_model_size])
+        output = self.dense(original_size_attention)
+
+        outputs = (output, present)
+        if output_attentions:
+            outputs = outputs + (attn, )
+        return outputs
+
+
+class EncoderLayer(nn.Layer):
+    def __init__(self, d_model_size, num_heads, dff, rate=0.1, epsilon=1e-6):
+        super().__init__()
+
+        self.multi_head_attention = MultiHeadAttention(d_model_size, num_heads)
+        self.ffn = nn.Sequential(
+            nn.Linear(d_model_size, dff),
+            nn.ReLU(), nn.Linear(dff, d_model_size))
+        self.layernorm1 = nn.LayerNorm(d_model_size, epsilon=epsilon)
+        self.layernorm2 = nn.LayerNorm(d_model_size, epsilon=epsilon)
+
+        self.dropout1 = nn.Dropout(rate)
+        self.dropout2 = nn.Dropout(rate)
+
+    def forward(
+            self,
+            x,
+            mask,
+            layer_past=None,
+            attention_mask=None,
+            use_cache=False,
+            output_attentions=False, ):
+        normed = self.layernorm1(x)
+        attn_outputs = self.multi_head_attention(
+            normed,
+            normed,
+            normed,
+            mask,
+            layer_past=layer_past,
+            attention_mask=attention_mask,
+            use_cache=use_cache,
+            output_attentions=output_attentions, )
+        attn_output = attn_outputs[0]
+        attn_output = self.dropout1(attn_output)
+        out1 = x + attn_output
+
+        out2 = self.layernorm2(out1)
+        ffn_output = self.ffn(out2)
+        ffn_output = self.dropout2(ffn_output)
+        out2 = out1 + ffn_output
+
+        outputs = (out2, ) + attn_outputs[1:]
+        return outputs
+
+
+class CTRLPreTrainedModel(PretrainedModel):
+    base_model_prefix = "transformer"
```

Comment:
The `base_model_prefix` for CTRL should be 'ctrl'

Reply:
done